replit_river/client_session.py

```diff
@@ -100,29 +100,26 @@ async def send_upload(
         self._streams[stream_id] = output
         first_message = True
         try:
-            if init and init_serializer:
-                await self.send_message(
-                    stream_id=stream_id,
-                    control_flags=STREAM_OPEN_BIT,
-                    service_name=service_name,
-                    procedure_name=procedure_name,
-                    payload=init_serializer(init),
-                )
-                first_message = False
-            # If this request is not closed and the session is killed, we should
-            # throw exception here
-            async for item in request:
-                control_flags = 0
-                if first_message:
-                    control_flags = STREAM_OPEN_BIT
-                    first_message = False
-                await self.send_message(
-                    stream_id=stream_id,
-                    service_name=service_name,
-                    procedure_name=procedure_name,
-                    control_flags=control_flags,
-                    payload=request_serializer(item),
-                )
+            await self.send_message(
+                stream_id=stream_id,
+                control_flags=STREAM_OPEN_BIT,
+                service_name=service_name,
+                procedure_name=procedure_name,
+                payload=init_serializer(init),
+            )
+            first_message = False
```

Comment:
`first_message` should be no longer needed since we always send a `STREAM_OPEN_BIT` and we can remove the bit from `send_close_stream`

Reply:
Based on how the code worked previously, if we had an exception during sending the first message, then `STREAM_OPEN_BIT` was sent in the `send_close_stream` -- was that not intentional?